core-backend/src/runtime.rs

```diff
@@ -160,4 +164,20 @@ where
             },
         }
     }
+
+    pub fn check_func_forbiddenness(&mut self, syscall_name: SyscallName) -> Result<(), HostError> {
+        if self.ext_mut().forbidden_funcs().contains(&syscall_name)
+            || self
+                .ext_mut()
+                .endpoint_dispatch_kind()
```

Comment:
doesn't it make more sense from where we take this value?
```suggestion
                .ext_mut()
                .msg_ctx()
                .dispatch_kind()
                .forbids(x)
```

Reply:
it's not like it belongs to endpoint

Reply:
Agree